src/sentry/integrations/grpc/services/scm_service.py

```diff
@@ -0,0 +1,641 @@
+                    if return_type:
+                        return return_type()
+                    else:
+                        return empty_pb2.Empty()
+
+                try:
+                    # Transform request if transformer provided
+                    if transform_request:
+                        args, kwargs = transform_request(request)
+                    else:
+                        # Default: pass empty args and kwargs
+                        args, kwargs = [], {}
+
+                    # Call the integration method
+                    result = getattr(installation, actual_method_name)(*args, **kwargs)
+
+                    # Transform response if transformer provided
+                    if transform_response:
+                        return transform_response(result)
+                    else:
+                        # Default: return result as-is (should be a protobuf message)
+                        return result
+
+                except Exception as e:
+                    logger.exception("Error calling %s", actual_method_name)
+                    context.set_code(grpc.StatusCode.INTERNAL)
+                    context.set_details(str(e))
+                    return_type = func.__annotations__.get("return")
+                    if return_type:
+                        return return_type()
+                    else:
+                        return empty_pb2.Empty()
+
+        # Copy annotations from original function to wrapper
+        wrapper.__annotations__ = func.__annotations__.copy()
+        return wrapper
+
+    return decorator
+
+
+class ScmServicer(scm_pb2_grpc.ScmServiceServicer, BaseGrpcServicer):
+    """
+    gRPC service that generically proxies requests to integration methods.
+
+    This service uses a context manager to load integrations and a generic
+    wrapper to call integration methods, eliminating boilerplate.
+    """
+
+    @contextmanager
+    def integration_installation(self, request: Any, context: grpc.ServicerContext):
+        """
+        Context manager that loads the integration installation and handles errors.
+
+        Args:
+            request: The gRPC request object (must have organization_id)
+            context: The gRPC context
+
+        Yields:
+            The integration installation instance
+        """
+        try:
+            # Get organization
+            org = Organization.objects.get(id=request.organization_id)
+
+            # Get integration based on what's in the request
+            if hasattr(request, "integration_id") and request.integration_id:
+                org_integration = integration_service.get_organization_integration(
+                    organization_id=org.id, integration_id=request.integration_id
+                )
+            elif hasattr(request, "provider") and request.provider != scm_pb2.PROVIDER_UNSPECIFIED:
+                provider_map = {
+                    scm_pb2.PROVIDER_GITHUB: "github",
```

Comment:
If we keep these 1:1 with the our `IntegrationProviderSlug` enum, this mapping becomes fairly trivial: https://github.com/getsentry/sentry/blob/6469adef34055d0c1da56a90d684bc27ac250b83/src/sentry/integrations/types.py#L32